lib/linalg/NNLS.cpp

```diff
@@ -58,14 +58,15 @@ extern "C" {
 NNLSSolver::NNLSSolver(double const_tol, int min_nnz, int max_nnz,
                        int verbosity,
                        double res_change_termination_tol,
-                       double zero_tol, int n_outer, int n_inner)
+                       double zero_tol, int n_outer, int n_inner,
+                       const NNLS_termination criterion)
```

Comment:
It might be good to have a `CAROM_VERIFY` check in the constructor to make sure that `NNLS_termination` is set to `LINF` or `L2`.